content/en/docs/concepts/security/service-accounts.md

```diff
@@ -0,0 +1,200 @@
+the `automountServiceAccountToken` field in your Pod specification to `false`.
+
+## Use cases for Kubernetes service accounts
+
+As a general guideline, you can use service accounts to provide identities in
+the following scenarios:
+
+* Your Pods need to communicate with the Kubernetes API server, for example in
+  situations such as the following:
+  *  Providing read-only access to sensitive information stored in Secrets.
+  *  Granting [cross-namespace access](TODO), such as allowing a Pod in namespace
+     `example` to read, list, and watch for Lease objects in the
+     `kube-node-lease` namespace.
+* Your Pods need to communicate with an external service. For example,
+  [Authenticating to a private image registry using an `imagePullSecret`](/docs/tasks/configure-pod-container/configure-service-account/#add-imagepullsecrets-to-a-service-account).
+* An external service needs to communicate with the Kubernetes API server. For
+  example, authenticating to the cluster as part of a CI/CD pipeline.
+
+## How to use service accounts
+
+To use a Kubernetes service account, you do the following:
+
+1. Create the ServiceAccount object in the Kubernetes API using a Kubernetes
+   client like `kubectl` or a manifest that defines the object.
+1. Grant permissions to the ServiceAccount object using an authorization
+   mechanism such as RBAC.
+1. Assign the ServiceAccount object to Pods during Pod creation.
+
+For instructions, refer to
+[Configure Service Accounts for Pods](/docs/tasks/configure-pod-container/configure-service-account/).
+
+### Grant permissions to a ServiceAccount
+
+You can use the built-in Kubernetes
+[role-based access control (RBAC)](/docs/reference/access-authn-authz/rbac/)
+mechanism to grant the minimum permissions required by each service account.
+You create a *role*, which grants access, and then *bind* the role to your
+ServiceAccount. RBAC lets you define a minimum set of permissions so that the
+service account permissions follow the principle of least privilege. Pods that
+use that service account don't get more permissions than are required to
+function correctly.
+
+#### Cross-namespace access using a ServiceAccount
+
+You can use RBAC to allow service accounts in one namespace to perform actions
+on resources in a different namespace in the cluster. For example, consider a
+scenario where you have a service account and Pod in the `dev` namespace and
+you want your Pod to see Jobs running in the `batch` namespace. You could
+create a Role object that grants permissions to list Job objects. Then,
+you'd create a RoleBinding object in the `batch` namespace to bind the Role
+to the ServiceAccount object. Now, Pods in the `dev` namespace can list Job
+objects in the `batch` namespace using that service account.
+
+### Use ServiceAccount credentials from a Pod
+
+To assign a service account to a Pod, you provide the Pod with the credentials
+for the service account, which the Pod can then use to authenticate with the
```

Comment:
from a user-perspective, you assign a service account to a pod by setting `spec.serviceAccountName`; the system does the injection (using one of the following methods) automatically. I probably wouldn't mention any of the specific methods this in this section... it's more relevant in a "Use ServiceAccount credentials outside a Pod" section

Reply:
Ah I see. https://kubernetes.io/docs/tasks/configure-pod-container/configure-service-account/#manually-create-an-api-token-for-a-serviceaccount cleared it up a bit. So depending on k8s version, the method by which Kubernetes gets SA credentials is different? In versions earlier than 1.22, tokens were autogenerated for Secret-based mounting. In 1.22 and later, Kubernetes gets the tokens using TokenRequest and mounts as projected volumes. When would you use those manually then? tokenrequest I understand, to get SA credentials in an application, but what about manually mounting as a projected volume? 

Sorry @liggitt I dunno if my response is very coherent

Reply:
> I'd still need to do something in my Pod spec to tell Kubernetes that I want a projected volume, for example, right?

No, auto-injection of a token for the pod's service account silently switched mechanisms from secret-based to projected-volume-based in ~1.21/1.22.

> Or is it automatic based on version and the only thing I need to set is the ServiceAccount name?

Yeah, pod authors just set the spec.serviceAccountName, API server admission handles the rest of the volume setup.

> When would you use those manually then? tokenrequest I understand, to get SA credentials in an application, but what about manually mounting as a projected volume?

If you want a service account token mounted at a non-standard location, or with an audience other than the kube-apiserver, you would manually set up a projected token mount. That would let you get a token you could present to some other audience (for example, scenarios mentioned in https://github.com/cert-manager/cert-manager/issues/4144#issuecomment-930087333)

Reply:
I'll fiddle with this today, this is the last bit I think is needed to get this PR in good order